### PR TITLE
Add app loader

### PR DESCRIPTION
**Pull request by eatonphil**
screen shot 2017-05-24 at 10 13 47 am

### CODE REVIEW

**Comment by abemassry** [comment]
I'm a fan of a loading screen in general as opposed to showing a blank screen for anything > 4 sec

File: scss/components/AppLoader.scss [hunk 1]
```diff
@@ -0,0 +1,97 @@
+// Source: https://github.com/lukehaas/css-loaders
```
Comment (mparke) [inline scss/components/AppLoader.scss]: can be included via npm?

Comment (eatonphil) [inline scss/components/AppLoader.scss]: Yes, but it's a pretty small dependency and I needed to make some changes to it.

Comment (eatonphil) [inline scss/components/AppLoader.scss]: More importantly, it's not like we need to keep it up-to-date with upstream security / bug fixes sort of thing.

Comment (mparke) [inline scss/components/AppLoader.scss]: so, you forked it?

**Comment by mparke** [comment]
We could show this over the UI with an opacity:

screen shot 2017-05-24 at 2 24 37 pm

**Comment by eatonphil** [comment]
So long as response times are reasonable (not 10 seconds), I'd rather the app be fully ready before we show anything.

**Comment by phuegel** [comment]
The header (minus notification counts) should be static. Why can't they be loaded immediately while you wait for the body content below? Like the following (Blue constant & Red dynamic content waiting for complete responses):

image

**Comment by phuegel** [comment]
BTW - see the following for React loading screen example. 
https://stackoverflow.com/questions/40987309/react-display-loading-screen-while-dom-is-rendering

**Comment by mparke** [comment]
I agree w/ @phuegel that we should show the header UI at least while body loads.

**Comment by eatonphil** [comment]
The most we can do here is display the header with the Linode logo (and the three main nav items). However, it is not possible to show the rest of the items because (like the body), they are not returned immediately. I'll make a commit to add the header and the Linode logo for now. But I don't think it's appropriate to show the header nav items because those are clearly "logged in" items and it isn't clear you aren't entirely "logged in" until we can display your username at least.

**Comment by eatonphil** [comment]
Updated with some nice improvements:

screen shot 2017-05-26 at 1 38 28 pm

Improvements include:

* Moved all session related logic into src/sessions.js alone
* Make authenticated calls first in src/layouts/Layouts.js so the redirect to login gets triggered sooner
* If a request fails with 401, remove the current token from localstorage so if you happen to not log in immediately and refresh the page, no time is wasted redirecting you to login.

**Comment by mparke** [comment]
@eatonphil build failing?

**Pull request merged**